tasks/main.yml

```diff
@@ -15,25 +11,76 @@
     - "{{ ansible_facts['distribution'] ~ '_' ~
           ansible_facts['distribution_version'] }}.yml"
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __crypto_policies_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __crypto_policies_vars_file is file
+
+- name: OS is supported
+  meta: end_host
+  when:
+    - not __crypto_policies_os_supported|bool
 
-# Examples of some tasks:
 - name: Ensure required packages are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __crypto_policies_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: yes
-  loop: "{{ __template_services }}"
-
-- name: Generate /etc/{{ __template_foo_config }}
-  template:
-    src: "{{ __template_foo_config }}.j2"
-    dest: /etc/{{ __template_foo_config }}
-    backup: yes
-  notify: template handler to restart services
+- include_tasks: gather_facts.yml
+
+- name: Find out what base policies are available
+  shell: >
+    set -euo pipefail && (
+      ls -q /usr/share/crypto-policies/policies/*pol;
+      ls -q /etc/crypto-policies/policies/*pol || true
+    ) 2>/dev/null | xargs -n 1 basename | cut -d '.' -f1 |  sort | uniq
+  changed_when: false
+  register: __crypto_policies_available_policies
+
+- name: Set available base policies fact
+  set_fact:
+    crypto_policies_available_policies: >-
+      {{ __crypto_policies_available_policies.stdout_lines }}
+    cacheable: yes
+
+- name: Find out what policy modules are available
+  shell: >
+    set -euo pipefail && (
+      ls -q /usr/share/crypto-policies/policies/modules/*pmod;
+      ls -q /etc/crypto-policies/policies/modules/*pmod || true
+    ) 2>/dev/null | xargs -n 1 basename | cut -d '.' -f1 |  sort | uniq
```

Comment:
This is something we would like to ideally hide inside of the `crypto-policies` scripts as it is ugly and error prone, but it is unlikely to be available in RHEL 8.4 and before so we will need an implementation here too.  Or would it make sense to write for this a python module in this role?

Reply:
I think it is ok - but you should use the `find` module instead of using `shell` with `ls` - for this and the base policies - see https://docs.ansible.com/ansible/2.9/modules/find_module.html#find-module

Reply:
Thanks for suggestion. I did not know how to capture this more in Ansible way, but after some reading through Jinja2 filters documentation I manage to come up with nicer solution (57a9f0e3). Please, let me know if you see there some room for improvement.